src/SwedbankPay.Sdk/SwedbankPayClient.cs

```diff
@@ -1,44 +1,60 @@
+        private readonly HttpClient httpClient;
+
+        public SwedbankPayClient(HttpClient httpClient, IPaymentOrdersResource paymentOrders, IConsumersResource consumers, IPaymentsResource payments)
         {
             if (!ServicePointManager.SecurityProtocol.HasFlag(SecurityProtocolType.Tls12))
                 ServicePointManager.SecurityProtocol |= SecurityProtocolType.Tls12;
-            if (httpClient == null)
-                throw new ArgumentNullException(nameof(httpClient));
 
-            if (httpClient.BaseAddress == null)
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+
+            if (this.httpClient.BaseAddress == null)
                 throw new ArgumentNullException(nameof(httpClient), $"{nameof(httpClient.BaseAddress)} cannot be null.");
 
-            if (httpClient.DefaultRequestHeaders?.Authorization?.Parameter == null)
+            if (this.httpClient.DefaultRequestHeaders?.Authorization?.Parameter == null)
                 throw new ArgumentException($"Please configure the {nameof(httpClient)} with an Authorization header.");
 
-            var swedbankLogger = logger ?? NullLogger.Instance;
-            var swedbankPayHttpClient = new SwedbankPayHttpClient(httpClient, swedbankLogger);
-            PaymentOrder = new PaymentOrderResource(swedbankPayHttpClient);
-            Consumers = new ConsumersResource(swedbankPayHttpClient);
-            Payment = new PaymentsResource(swedbankPayHttpClient);
+            PaymentOrders = paymentOrders;
+            Consumers = consumers;
+            Payments = payments;
         }
 
+        public SwedbankPayClient(HttpClient httpClient)
+        {
+            if (!ServicePointManager.SecurityProtocol.HasFlag(SecurityProtocolType.Tls12))
+                ServicePointManager.SecurityProtocol |= SecurityProtocolType.Tls12;
 
-        public IPaymentOrderResource PaymentOrder { get; }
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
 
-        public IConsumersResource Consumers { get; }
+            if (this.httpClient.BaseAddress == null)
+                throw new ArgumentNullException(nameof(httpClient), $"{nameof(httpClient.BaseAddress)} cannot be null.");
+
+            if (this.httpClient.DefaultRequestHeaders?.Authorization?.Parameter == null)
+                throw new ArgumentException($"Please configure the {nameof(httpClient)} with an Authorization header.");
 
-        public IPaymentsResource Payment { get; }
+            var cardPaymentsResource = new CardPaymentsResource(this.httpClient);
+            var swishPaymentsResource = new SwishPaymentsResource(this.httpClient);
+
+            PaymentOrders = new PaymentOrdersResource(this.httpClient);
+            Consumers = new ConsumersResource(this.httpClient);
+            Payments = new PaymentsResource(this.httpClient, cardPaymentsResource, swishPaymentsResource);
```

Comment:
Can `PaymentsResource` provide a constructor that initializes `CardPaymentsResource` and `SwishPaymentsResource` itself?